### ISSUE TITLE
Some apps crossing over screen edges

### ISSUE THREAD

**Issue by Gunni** [digitalcreations/MaxTo#115]
It's hard to explain and it only happens in Babun and KiTTY.

![screenie-1447756645](https://cloud.githubusercontent.com/assets/358099/11209295/23e0f808-8d18-11e5-8dfd-ac8f5ba9b2bf.jpg)

Left screen is vertical, center and right screen are horizontal.

Left screen has been split once horizontally.

When i use shift+maximize on windows in regions they fill their section but go a bit onto the center screen, actually perfectly hiding their scrollbar on the other screen...

**Comment by vegardlarsen** [comment]
This problem seems similar to one I've seen with `mintty`. PuTTY works well with MaxTo, so it has to be something special that KiTTY does that PuTTY doesn't.

My guess is that they are doing something special with their scrollbar. Can you check to see if it has an option to turn of the scrollbar (you can Shift+PageUp/PageDn to scroll), and see if that makes it line up perfectly?

**Comment by Gunni** [comment]
@vegardlarsen yes it has that option and yes then the window fits perfectly...

**Comment by vegardlarsen** [comment]
I'll be looking into this more thoroughly, but then at least you have a workaround for now. I may just have to add a list of software that needs a special offset rectangle, as there seems to be 3 programs now that are misbehaving.

**Comment by Gunni** [comment]
Thanks! :)

**Comment by reidrankin** [comment]
I'm getting this problem with standard PuTTY, version 0.66, and the cygwin mintty windows.

**Comment by saivert** [comment]
I suspect this is because mintty readjusts the window after resize to fit the character grid of the terminal. It only resizes in steps as you can see if you freely resize it with the mouse.
Has there been tests with other software (other than terminal emulators) that seem to have special resize handling?

**Comment by vegardlarsen** [comment]
With `mintty`, can you check to see if it has an option to remove the scrollbar (as putty does)?

I have had a browse through their source code now, and it appears they will call `win_adapt_term_size` ([link](https://github.com/mintty/mintty/blob/master/src/winmain.c#L1780)), where they do:

```c
 win_set_pixels(term.rows * cell_height, term.cols * cell_width);
```

under certain conditions. I have not had a chance to look very detailed into this, but this should help me get started when I get to examine this more closely.

**Comment by Gunni** [comment]
I'm abandoning this ticket as i no longer use MaxTo as it started scaling text up when i used MaxTo to resize windows.

**Comment by skewty** [comment]
Can this please be re-opened as I affected by this issue using putty 0.70.

**Comment by vegardlarsen** [comment]
Yes. I am reopening this, but I am not sure it will be possible to fix.

**Comment by JFouts** [comment]
Been an open issue for 4 years, but for what it is worth I would like to see this issue fixed as well.
Not just putty that is affected. Command Prompt, Powershell and Cygwin all have the same issue. Clearly something functions differently for windows terminals. If you could figure out want it is that would be awesome.

**Comment by vegardlarsen** [comment]
With PuTTy, the issue disappears if you disable the scrollbar. This is not an option with the new "console host" in Windows 10.

The issue is related to the fact that console windows now appears to have a different window size than what it appears they do; as the scroll bar is "invisible" to MaxTo. And this wrong size is observable not only for MaxTo, but also in Spy++ and similar tools. I can't quite figure out what they are doing.

**Comment by Etherealflux** [comment]
Would it be practical to add a "trim" for the size of each window type? For example, Powershell would have its size set to, say, 20px less than the region it's being placed in, which would account for the annoying scrollbar behavior.

It's a bit odd that this isn't an issue with vanilla window snapping.

**Comment by vegardlarsen** [comment]
@Etherealflux That actually makes a lot of sense. We could do this as an extension of the current compatibility shim system.

**Comment by rooby** [comment]
Git bash is another one where scroll bars go out of the region.

**Comment by rooby** [comment]
One possible solution could be to have a list of programs that adjust their width based on the scroolbar width.
Ideally a list that the user can modify, but it could have known problem applications as a default.
Could be a setting you could disable if you wanted to retain the current behaviour.

**Comment by rooby** [comment]
It seems as though Microsoft's [Power Toys'](https://github.com/microsoft/PowerToys) FancyZones doesn't have this problem, at least for the programs I use.
Their code might have some useful insights.
